cmd/soroban-cli/src/commands/contract/arg_parsing.rs

```diff
@@ -246,3 +246,27 @@ pub fn output_to_string(
     }
     Ok(TxnResult::Res(res_str))
 }
+
+fn resolve_sc_address(
+    addr_or_alias: &str,
+    config: &config::Args,
+) -> Result<(String, Option<SigningKey>), Error> {
+    let addr_or_alias = addr_or_alias.trim_matches('"');
+    let sc_address: ScAddress = addr_or_alias.parse().unwrap();
+    let account = match sc_address {
+        ScAddress::Address(addr) => return Ok((addr.to_string(), None)),
+        addr @ ScAddress::Alias(_) => {
+            let addr = addr.resolve(&config.locator, &config.get_network()?.network_passphrase)?;
+            match addr {
+                xdr::ScAddress::Account(account) => account.to_string(),
+                contract @ xdr::ScAddress::Contract(_) => return Ok((contract.to_string(), None)),
+            }
+        }
+    };
+    let cmd = crate::commands::keys::address::Cmd {
+        name: addr_or_alias.to_string(),
+        hd_path: Some(0),
+        locator: config.locator.clone(),
+    };
+    Ok((account, cmd.private_key().ok()))
```

Comment:
It looks like this function is really two distinct pieces of logic bundled together. Suggest breaking it up accordingly. I pushed a change to do that in 2bf2a7a96038768a986ffadc1bf6648595b8692a, but feel free to revert it if I'm missing something.